back/app/services/export/geojson/geojson_generator.rb

```diff
@@ -10,7 +10,7 @@ def initialize(phase, field)
         @phase = phase
         @field = field
         @inputs = phase.ideas.native_survey.published
-        @fields_in_form = IdeaCustomFieldsService.new(phase.custom_form).reportable_fields
+        @fields_in_form = IdeaCustomFieldsService.new(phase.custom_form).all_fields.filter(&:supports_xlsx_export?)
```

Comment:
Why I changed this:
1. I wanted to do a much bigger refactoring, with single table inheritance on the input_type. This would make adding new fields easier in the future, because most of the behaviour would be defined by the subclass for that field type. In this case, it's easier to remember to specify `supports_xlsx_export?` than to remember that we might want to change the `reportable_fields` (and forgetting this can result in bugs). In general, I think we should start moving as much input_type-specific behaviour into the custom field model, to reduce coupling and increase cohesion.
2. Although I like the name `reportable_fields`, I think it's not clear which exact features are using this. I think defining individual `supports_<feature>?` predicates is much more clear in that regard.

Reply:
Alright, although I think we should get rid of the idea_custom_fields_service in the long run, I added the two methods you suggested for xlsx and geojson.